pilot/pkg/kube/inject/inject.go

```diff
@@ -742,3 +742,13 @@ func sidecarTemplateVersionHash(in string) string {
 	hash := sha256.Sum256([]byte(in))
 	return hex.EncodeToString(hash[:])
 }
+
+func potentialPodName(metadata *metav1.ObjectMeta) string {
+	if metadata.Name != "" {
+		return metadata.Name
+	}
+	if metadata.GenerateName != "" {
+		return metadata.GenerateName + "*****"
```

Comment:
The **** will look a bit mysterious in the logs - maybe change to "(from GenerateName)" or something like that.

Reply:
@ostromart I think `GenerateName` is not widely known. `Deployments` on the other hand are.

What about `pod-prefix-***** (actual name not yet known)`?

Sorry, I'm lacking ideas :)

Reply:
Sure, sgtm.